### PR TITLE
[TrimmableTypeMap] Add ManifestGenerator

### PR DESCRIPTION
**Pull request by simonrozsival**
## Summary

Cecil-free Android manifest generation from component attributes captured by the SRM-based `JavaPeerScanner`. Part of #10807.

All code lives in the `Microsoft.Android.Sdk.TrimmableTypeMap` assembly (netstandard2.0) — no MSBuild dependency.

Depends on #10967. Used by #10980.

### Component attributes
- `[Activity]`, `[Service]`, `[BroadcastReceiver]`, `[ContentProvider]` with all named properties
- `MainLauncher` → `<intent-filter>` with MAIN/LAUNCHER + `android:exported="true"`
- `[IntentFilter]` → `<intent-filter>` with actions, categories, data elements
- `[MetaData]` → `<meta-data>` with name/value/resource
- `[Application]` → updates `<application>` element attributes
- `[Instrumentation]` → `<instrumentation>` under `<manifest>`

### Assembly-level attributes
- `[Permission]`, `[PermissionGroup]`, `[PermissionTree]` → `<permission>` etc. with dedup
- `[UsesPermission]` → `<uses-permission>` with maxSdkVersion
- `[UsesFeature]` → `<uses-feature>` (named + GL ES version variants)
- `[UsesLibrary]`, `[UsesConfiguration]` → inside `<application>`
- `[assembly: Application]` → merged into `<application>` attributes
- `[assembly: MetaData]`, `[assembly: Property]` → inside `<application>`

### Build properties
- `ManifestPlaceholders` — `${key}` → value replacement
- `CheckedBuild` → `android:debuggable` + `android:extractNativeLibs`
- `ApplicationJavaClass` → `<application android:name="...">`
- `VersionCode` defaults to `"1"`, `VersionName` defaults to `"1.0"` (matching legacy)

### Validation
- XA4213: component types must have a public parameterless constructor
- Duplicate Application type detection (assembly-level vs type-level)

### Runtime provider
- `MonoRuntimeProvider` / `NativeAotRuntimeProvider` injection
- Multi-process support (per-process providers)
- Dedup check against template (won't duplicate existing providers)

## Tests
- **30 test cases** (xunit, `[Fact]`/`[Theory]`), ~130ms
- Covers all component types, assembly-level attrs, template merging, dedup, placeholders, validation, enum conversion

### CODE REVIEW

File: src/Microsoft.Android.Sdk.TrimmableTypeMap/Generator/ManifestGenerator.cs [hunk 1]
```diff
@@ -0,0 +1,911 @@
...
+				element.SetAttributeValue (AndroidNs + m.XmlAttributeName, b ? "true" : "false");
+				break;
+			case MappingKind.Enum when m.EnumConverter is not null:
+				int intValue = value switch { int i => i, long l => (int)l, short s => s, byte b => b, _ => 0 };
+				if (intValue != 0) {
+					var strValue = m.EnumConverter (intValue);
+					if (strValue is not null) {
+						element.SetAttributeValue (AndroidNs + m.XmlAttributeName, strValue);
+					}
+				}
```
[the remark this answers is not in the document]
Comment (simonrozsival) [reply src/Microsoft.Android.Sdk.TrimmableTypeMap/Generator/ManifestGenerator.cs]: Fixed — removed `intValue != 0` guard in `ApplyMappings`. Now always calls `EnumConverter` and only sets the attribute when it returns non-null.

File: src/Microsoft.Android.Sdk.TrimmableTypeMap/Generator/ManifestGenerator.cs [hunk 2]
```diff
@@ -0,0 +1,911 @@
...
+			short s => s,
+			byte b => b,
+			_ => 0,
+		};
+		if (intValue != 0) {
+			var strValue = converter (intValue);
+			if (strValue is not null) {
+				element.SetAttributeValue (AndroidNs + xmlAttrName, strValue);
+			}
+		}
```
[the remark this answers is not in the document]
Comment (simonrozsival) [reply src/Microsoft.Android.Sdk.TrimmableTypeMap/Generator/ManifestGenerator.cs]: Fixed — same approach in `MapDictionaryEnumProperty`. Removed `intValue != 0` guard.

File: src/Microsoft.Android.Sdk.TrimmableTypeMap/Generator/ManifestGenerator.cs [hunk 3]
```diff
@@ -0,0 +1,911 @@
...
+		string packageName = "mono";
+		string className = "MonoRuntimeProvider";
+
+		if (string.Equals (AndroidRuntime, "nativeaot", StringComparison.OrdinalIgnoreCase)) {
+			packageName = "net.dot.jni.nativeaot";
+			className = "NativeAotRuntimeProvider";
+		}
+
+		app.Add (CreateRuntimeProvider ($"{packageName}.{className}", null, --appInitOrder));
+
```
[the remark this answers is not in the document]
Comment (simonrozsival) [reply src/Microsoft.Android.Sdk.TrimmableTypeMap/Generator/ManifestGenerator.cs:210]: Fixed — now checks for existing provider with matching `android:name` or `.__mono_init__` authorities before adding.

File: tests/Microsoft.Android.Sdk.TrimmableTypeMap.Tests/Generator/ManifestGeneratorTests.cs [hunk 4]
```diff
@@ -0,0 +1,614 @@
...
+
+		var meta2 = metaElements?.FirstOrDefault (m => (string?)m.Attribute (AndroidNs + "name") == "com.example.res");
+		Assert.NotNull (meta2);
+		Assert.Equal ("@xml/config", (string?)meta2?.Attribute (AndroidNs + "resource"));
+	}
+
+	[InlineData (ComponentKind.Service, "service")]
+	[InlineData (ComponentKind.BroadcastReceiver, "receiver")]
+	public void Component_BasicProperties (ComponentKind kind, string elementName)
+	{
```
[the remark this answers is not in the document]
Comment (simonrozsival) [reply tests/Microsoft.Android.Sdk.TrimmableTypeMap.Tests/Generator/ManifestGeneratorTests.cs:191]: Fixed — added `[Theory]` attribute.

File: tests/Microsoft.Android.Sdk.TrimmableTypeMap.Tests/Generator/ManifestGeneratorTests.cs [hunk 5]
```diff
@@ -0,0 +1,614 @@
...
+		var app = doc.Root?.Element ("application");
+
+		Assert.Equal ("false", (string?)app?.Attribute (AndroidNs + "allowBackup"));
+		Assert.Equal ("@mipmap/ic_launcher", (string?)app?.Attribute (AndroidNs + "icon"));
+	}
+
+	[InlineData ("", "", "1", "1.0")]
+	[InlineData ("42", "2.5", "42", "2.5")]
+	public void VersionDefaults (string versionCode, string versionName, string expectedCode, string expectedName)
+	{
```
[the remark this answers is not in the document]
Comment (simonrozsival) [reply tests/Microsoft.Android.Sdk.TrimmableTypeMap.Tests/Generator/ManifestGeneratorTests.cs:326]: Fixed — added `[Theory]` attribute.

File: tests/Microsoft.Android.Sdk.TrimmableTypeMap.Tests/Generator/ManifestGeneratorTests.cs [hunk 6]
```diff
@@ -0,0 +1,614 @@
...
+
+		Assert.Equal ("24", (string?)usesSdk?.Attribute (AndroidNs + "minSdkVersion"));
+		Assert.Equal ("34", (string?)usesSdk?.Attribute (AndroidNs + "targetSdkVersion"));
+	}
+
+	[InlineData (true, false, false, "debuggable", "true")]
+	[InlineData (false, true, false, "debuggable", "true")]
+	[InlineData (false, false, true, "extractNativeLibs", "true")]
+	public void ApplicationFlags (bool debug, bool forceDebuggable, bool forceExtractNativeLibs, string attrName, string expected)
+	{
```
[the remark this answers is not in the document]
Comment (simonrozsival) [reply tests/Microsoft.Android.Sdk.TrimmableTypeMap.Tests/Generator/ManifestGeneratorTests.cs:356]: Fixed — added `[Theory]` attribute.

File: src/Microsoft.Android.Sdk.TrimmableTypeMap/Generator/ManifestModel.cs [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+{
+	public IReadOnlyList<string> Actions { get; set; } = Array.Empty<string> ();
+	public IReadOnlyList<string> Categories { get; set; } = Array.Empty<string> ();
+	public Dictionary<string, object?> Properties { get; set; } = new Dictionary<string, object?> ();
+}
+
+public class MetaDataInfo
+{
+	public string Name { get; set; } = "";
+	public string? Value { get; set; }
```
Comment (simonrozsival) [inline src/Microsoft.Android.Sdk.TrimmableTypeMap/Generator/ManifestModel.cs:36]: 🤖 💡 **Formatting** — Per repo conventions, use `[]` instead of `Array.Empty<T>()`:

```csharp
public IReadOnlyList<IntentFilterInfo> IntentFilters { get; set; } = [];
public IReadOnlyList<MetaDataInfo> MetaData { get; set; } = [];
```

Same applies to `IntentFilterInfo.Actions`, `IntentFilterInfo.Categories` below.

_Rule: Use `[]` not `Array.Empty<T>()`_

**Review by simonrozsival — COMMENTED** [review]
## 🤖 AI Review Summary

**Verdict**: ✅ LGTM

Found **1 suggestion** (no blockers):

- 💡 **Formatting**: Use `[]` instead of `Array.Empty<T>()` in `ManifestModel.cs` (4 occurrences)

👍 Clean, well-structured manifest generator with good separation of concerns — enum conversion, property mapping, component building, and assembly-level elements are each in focused files. Excellent test coverage (30 cases) covering all component types, template merging, dedup, placeholders, enum flags, and edge cases (abstract types, existing components). Proper use of `CultureInfo.InvariantCulture` for numeric formatting. Data models are kept together in `ManifestModel.cs` which is pragmatic for 13 small DTOs.

**CI**: All checks pending (branch just rebased). Cannot verify build status yet.

---
_Review generated by android-reviewer from [review guidelines](https://github.com/dotnet/android/blob/main/.github/skills/android-reviewer/references/review-rules.md)._
